src/taipy/core/_scheduler/_scheduler.py

```diff
@@ -131,11 +141,10 @@ def _unlock_edit_on_outputs(jobs: Union[Job, List[Job], Set[Job]]):
 
     @classmethod
     def _on_status_change(cls, job: Job):
-        if job.is_completed():
+        if job.is_completed() or job.is_skipped():
             cls.__unblock_jobs()
-            cls.__check_and_execute_jobs_if_development_mode()
-        elif job.is_canceled():
-            cls.__check_and_execute_jobs_if_development_mode()
+        elif job.is_failed():
+            cls._fail_subsequent_jobs(job)
```

Comment:
When the job fails, I don't think we unlock the outputs. Do we?
Do we have a test that check that asserts that after a job fails its outputs are unlocked?

Reply:
As we discussed earlier, we're only unlocking the abandoned jobs and not the failed job. I've added some manipulation to the set to_fail_or_cancel_jobs to make sure we also unlock the failed job